android/sdk/src/main/java/org/jitsi/meet/sdk/AudioModeModule.java

```diff
@@ -318,31 +505,42 @@ private boolean updateAudioRoute(int mode) {
             return false;
         }
 
-        boolean useSpeaker = (mode == VIDEO_CALL);
-
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-            // On Android >= M we use the AudioDeviceCallback API, so turn on
-            // Bluetooth SCO from the start.
-            if (audioManager.isBluetoothScoAvailableOffCall()) {
-                audioManager.startBluetoothSco();
-            }
+        boolean bluetoothAvailable = availableDevices.contains(DEVICE_BLUETOOTH);
+        boolean earpieceAvailable = availableDevices.contains(DEVICE_EARPIECE);
+        boolean headsetAvailable = availableDevices.contains(DEVICE_HEADPHONES);
+
+        // Pick the desired device based on what's available and the mode.
+        String audioDevice;
+        if (bluetoothAvailable) {
+            audioDevice = DEVICE_BLUETOOTH;
+        } else if (headsetAvailable) {
+            audioDevice = DEVICE_HEADPHONES;
+        } else if (mode == AUDIO_CALL && earpieceAvailable) {
+            audioDevice = DEVICE_EARPIECE;
         } else {
-            // On older Android versions we must set the Bluetooth route
-            // manually. Also disable the speaker in that case.
-            setBluetoothAudioRoute(
-                    bluetoothHeadsetMonitor.isHeadsetAvailable());
-            if (bluetoothHeadsetMonitor.isHeadsetAvailable()) {
-                useSpeaker = false;
-            }
+            audioDevice = DEVICE_SPEAKER;
+        }
+
+        // Consider the user's selection
+        if (userSelectedDevice != null
+            && availableDevices.contains(userSelectedDevice)) {
```

Comment:
The above line needs +1 indent to the right in order to distinguish it from the next line.